core/vm/custom_eip.go

```diff
@@ -0,0 +1,152 @@
+package vm
+
+import (
+	"fmt"
+	"sort"
+	"strings"
+)
+
+// OpCodeInfo contains information required to identify an EVM operation.
+type OpCodeInfo struct {
+	Number OpCode
+	Name   string
+}
+
+// Operation is an utility struct that wraps the private type
+// operation.
+type Operation struct {
+	Op *operation
+}
+
+// ExtendActivators allows to merge the go ethereum activators map
+// with additional custom activators.
+func ExtendActivators(eips map[int]func(*JumpTable)) error {
+	// Catch early duplicated eip.
+	keys := make([]int, 0, len(eips))
+	for k := range eips {
+		if ValidEip(k) {
+			return fmt.Errorf("duplicate activation: %d is already present in %s", k, ActivateableEips())
+		}
+		keys = append(keys, k)
+	}
+
+	// Sorting keys to ensure deterministic execution.
+	sort.Ints(keys)
+
+	for _, k := range keys {
+		activators[k] = eips[k]
+	}
+	return nil
+}
+
+// GetActivatorsEipNumbers returns the keys of EIPs registered in
+// the activators map.
+// Used only in tests.
+func GetActivatorsEipNumbers() []int {
+	keys := make([]int, 0, len(activators))
+	for k, _ := range activators {
+		keys = append(keys, k)
+	}
+
+	sort.Ints(keys)
+	return keys
+}
+
+// ExtendOperations returns an instance of the new operation and register it in the list
+// of available ones.
+// Return an error if an operation with the same name is already present.
+// This function is used to prevent the overwrite of an already existent operation.
+func ExtendOperations(
+	opInfo OpCodeInfo,
+	execute executionFunc,
+	constantGas uint64,
+	dynamicGas gasFunc,
+	minStack int,
+	maxStack int,
+	memorySize memorySizeFunc,
+) (*Operation, error) {
+	opName := strings.ToUpper(strings.TrimSpace(opInfo.Name))
+	if err := extendOpCodeStringLists(opInfo.Number, opName); err != nil {
+		return nil, err
+	}
+
+	operation := newOperation(execute, constantGas, dynamicGas, minStack, maxStack, memorySize)
+	op := &Operation{operation}
+
+	return op, nil
+}
+
+// newOperation returns an instance of a new EVM operation.
+func newOperation(
+	execute executionFunc,
+	constantGas uint64,
+	dynamicGas gasFunc,
+	minStack int,
+	maxStack int,
+	memorySize memorySizeFunc,
+) *operation {
+	return &operation{
+		execute:     execute,
+		constantGas: constantGas,
+		dynamicGas:  dynamicGas,
+		minStack:    minStack,
+		maxStack:    maxStack,
+		memorySize:  memorySize,
+	}
+}
+
+// GetConstantGas return the constant gas used by the operation.
+func (o *operation) GetConstantGas() uint64 {
```

Comment:
Looks like most of this is unused so I'm fine keeping it in if we ever need to extend this, but I can't think of a case yet